addon/components/utils/address-form.ts

```diff
@@ -126,45 +112,57 @@ export default class extends Component<UtilsAddressFormArgs> {
     });
   }
 
-  private fillInAddress(place: any): void {
+  private initInputListeners(autocomplete: google.maps.places.Autocomplete, input: HTMLElement): void {
+    autocomplete.addListener('place_changed', () => {
+      const place = autocomplete.getPlace();
+      this.fillInAddress(place);
+    });
+    input.addEventListener('focusout', (event) => {
+      if ((<HTMLInputElement>event.target).value !== this.args.address.address1) {
+        (<HTMLInputElement>event.target).value = this.args.address.address1;
+      }
+    });
+  }
+
+  private fillInAddress(place: google.maps.places.PlaceResult): void {
     let address1: string = '';
     let zipcode: string = '';
     let city: string = '';
 
-    for (const component of place.address_components.reverse()) {
-      const componentType = component.types[0];
-
-      switch (componentType) {
-        case 'street_number': {
+    place.address_components!.reverse().map((component) => {
+      const componentType: string = component.types[0];
+      const mapper: { [key: string]: () => void } = {
```

Comment:
You are re-creating the `const mapper` on each cycle of the `.map()` loop above.
Move the mapper before the map call, and if needed use the param of each anonymous function to pass the value. 
```
const mapper = {
   street_number: (value) => { address1 = `${value} ${address1}`; },
   ...
}
```
Pretty sure you can keep things inline so that we gain space. (this is currently longer than the switch case 😅 )
then pass `component.long_name` when using `mapper[componentType]?.(component.long_name)`
( use if (country) to pass the component.short_name for this one only )

Reply:
You could even prevent the `if (country) mapper[componentType]?.(component.short_name)` by passing `component`to the mapper input directly.
```
const mapper = {
   street_number: (component) => { address1 = `${component.long_name} ${address1}`; },
   ...
   country: (component) => {
     const selectedCountry = this.countries.find((c) => c.alpha2 === component.short_name);
     if (!selectedCountry) return;
     this.applyCountry(selectedCountry);
   }
}
```
Then usage:
```
place.address_components!.reverse().map((component) => {
   const componentType: string = component.types[0];
   mapper[componentType]?.(component);
}
```